carbon-design-system/carbon-addons-iot-react
# feat(table): add icon button to filter row

Closes #3688 

**Summary**

- Add icon button to filter row

**Change List (commits, features, bugs, etc)**

- Update `FilterHeaderRow` component to display icon button

**Acceptance Test (how to verify the PR)**

- _Simple example_
- Go to [this story](https://deploy-preview-3699--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-filtering)
- In the knobs area enable `Display icon button in filter row`
- Verify that icon has appeared
- Resize window horizontally to display scrollbar
- Verify that icon is pinned

</br>

- _With column grouping_
- Go to [this story](https://deploy-preview-3699--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--playground)
- In "Sort & filter" knobs enable `hasFilter: true` and `Display icon button in filter row`
- In "Column configuration" enable `Demo assign columns to groups`
- Click on filter icon to toolbar
- Verify that icon button in filter row has appeared
- Resize window horizontally, verify that icon is always visible

</br>

- _With row actions_
- In the same story
- In "Selections & actions" enable `hasRowActions`
- Click on filter icon to toolbar
- Verify that icon button in filter row has appeared
- Resize window horizontally, verify that icon is always visible

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
